### ISSUE TITLE
Concurrent operations possible ?

### ISSUE THREAD

**Issue by PandemiK911** [elemental-lf/benji#79]
Hello,
We've started using benji for backuping up a Proxmox cluster with Ceph storage.
I'm using postgresql as database, everything runs inside docker containers (for node independance).

On a node of the cluster, I'm mouting NFS file storage from a dedicated backup server.

My goal was to run backup once or twice a day on the Proxmox node and running batch deep-scrub directly on the backup server once a day with -P 15.
It *seems* that when deep-scrub runs in parallel of a backup, I end up with a lot of invalid versions. Running a complete deep-scrub (very long), rm and cleanup makes things right but I lose some backup versions.

So, which operations can be run concurrently ? Isn't backup and scrub supposed to be able to run at the same time ?

Thanks for your help and for this nice piece of code !
Best regards,
Julien

**Comment by elemental-lf** [comment]
Hey Julien,

operations on different versions should be able to run in parallel and locks should prevent any operations that are not supposed to run in parallel from doing so. Apart from that only one cleanup process can be running at a time, this is also protected by a lock. 

So that's the design and now we've got to figure out what's going wrong in your case:

* Could you please provide the information from the bug template? It can be found [here](https://github.com/elemental-lf/benji/blob/master/.github/ISSUE_TEMPLATE/bug_report.md).
* Could you also say something about the relation between the versions?  I mean are versions marked as invalid that have just been created? Or is there some other pattern between the versions which are marked as invalid and the currently running backups?

**Comment by PandemiK911** [comment]
Thanks for your answer.
I'll try to reproduce this 'corruption' on a limited subset of my data because running a full deep-scrub on our 6 TB data is a bit long ;-)
Just let me a few days to setup a parallel environnement and try to reproduce the problem.

Julien

**Comment by elemental-lf** [comment]
As no further feedback was provided I'm going to close this issue. If you stumble upon this problem again please open a new issue. Thanks!
